model/src/main/java/com/microsoft/java/bs/gradle/model/impl/DefaultGradleSourceSet.java

```diff
@@ -39,30 +39,18 @@ public class DefaultGradleSourceSet implements GradleSourceSet {
 
   private File sourceOutputDir;
 
-  private List<File> compileClasspath;
-
   private Set<File> resourceDirs;
 
   private File resourceOutputDir;
 
-  private File javaHome;
-
-  private String javaVersion;
-
-  private String gradleVersion;
-
-  private String sourceCompatibility;
-
-  private String targetCompatibility;
-
-  private List<String> compilerArgs;
-
   private Set<GradleModuleDependency> moduleDependencies;
 
   private Set<BuildTargetDependency> buildTargetDependencies;
 
   private boolean hasTests;
 
+  private Map<String, Object> extensions;
```

Comment:
It might be clearer to have a `LanguageExtension` interface that all the extensions inherit from, even if it is blank.  Then you're not using `Object` in collections and methods.

Reply:
Makes sense!

Reply:
I'm stuck here.

After change to `LanguageExtension` interface, the return of `gradleSourceSet.getExtensions().get("java")` becomes a proxy of `LanguageExtension`.

I cannot find a way to convert it to `JavaExtension`. class cast and reflection are not working.

Reply:
Ah - no problem then, I'd leave as is

Reply:
But I think your suggestion makes sense. I'll file an issue for this debt.

Reply:
Filed: https://github.com/microsoft/build-server-for-gradle/issues/128